### DISCUSSION TITLE
Can I use this code within a shopify app?

### DISCUSSION THREAD

**Discussion by vincenz2015** [add2cal/add-to-calendar-button#432] [Q&A] [answered]
I would like to use this within a shopify app, can I arrange some kind of agreement so that I can do that? (Not the builder) but just the button generation , I can make my own configurator.
Thanks

**Comment by jekuer** [comment, marked as answer]
Hi @vincenz2015 ,

Thank you for asking!
As always: It depends. 

**What would be fine:**
Offering an interface to enable users adding the free script to their shop website - for free "as it is" (which would then be basically the same as the user using the free script as described on our website).
What usually is also fine: Offering a solution, where it is not mainly about events and where the button plays a very minor role (like the streaming tool by @xoraingroup, mentioned at #334 ) (rule of thumb: solution is not focused on event management and it would work without advertising the button).

**What would not be allowed:** 
Building a plugin where users can create events, that are then integrated somewhere in the user's shop with the button (which would then be a managed solution for the script).
Generally speaking: Building a priced plugin, which basically mimics the functionality of our demo page is a no-go!
Same applies for any offering, which mainly consists of the Add to Calendar Button - like a Shopify plugin basically re-selling the open-source script.

So, short answer would be "unfortunately, not", based on the information we have.
Since it is also always about the (maybe confidential) details, I will contact you via LinkedIn to discuss.
